### PR TITLE
Feature/148 add browse page

### PR DESCRIPTION
**Pull request by isratc**
## Changes in this PR
create template markup
Add and remove class for the mobile structure
styles - hub section on the menu
style responsive views - and the gov website for the mobile list styling.
styles to match the prototype https://drive.google.com/drive/u/0/folders/1n7RZg2ItSgOV8f1DFRpk0wIPNo91CSoj
Trello https://trello.com/c/bId1W32a/217-style-browse-mega-menu-sub-list

## Screenshots of UI changes
Screenshot 2022-03-29 at 10 20 00

Screenshot 2022-03-29 at 10 20 27

Screenshot 2022-03-29 at 10 21 25

Screenshot 2022-03-29 at 10 21 56

### CODE REVIEW

File: cms/browse/views.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from cms.core.models import ExtendedMainMenuItem
+
+logger = logging.getLogger("general")
+
+
+def browse(request, programme, branch):
+    # we could get current site, query main menu for matching items and then
+    # load the items but we _know_ that ExtendedMenuItems will haveour list
+    # in it. 4+ queries -> 1 query
+    menu_items = ExtendedMainMenuItem.objects.all()
```
Comment (jacksonj04) [inline cms/browse/views.py:15]: This is true at the moment, but it's possible that we could add a second menu at some point which also used `ExtendedMainMenuItem`s, and this would explode in unexpected ways.

Is there a simple way to query for only the things on the menu?

File: cms/settings/base.py [hunk 2]
```diff
@@ -295,3 +296,6 @@
     ("footer-upper", "Footer (Upper links)"),
     ("footer-lower", "Footer (Lower links)"),
 )
+
+# what is the max length of mega menu captions in characters
+NHSEI_MAX_CATION_LENGTH = 50
```
[the remark this answers is not in the document]
Comment (jacksonj04) [reply cms/settings/base.py:305]: I'm ok with this being a constant rather than an env variable since it's pretty much bound by what can fit in the menu design.

File: cms/templates/partials/megamenu/megamenu.html [hunk 3]
```diff
@@ -11,7 +11,8 @@
     {#  Main menu conatiner - start #}
     <div class="main-menu-wrapper nhsuk-grid-column-three-quarters">
       <div class="main-menu">
-        <div><div class="programme_section">Programmes</div>{% main_menu max_levels=1 template="partials/megamenu/navigation.html" %}</div>
+        <div><div class="programme_section">Programmes</div>
```
Comment (jacksonj04) [inline cms/templates/partials/megamenu/megamenu.html:14]: I'm not completely happy with "programmes" being a title here, might need to go another round with content design to see if this means anything for site users, but leave it for now.

File: cms/templates/partials/megamenu/navigation.html [hunk 4]
```diff
@@ -1,9 +1,9 @@
-{% load menu_tags %}
+{% load menu_tags browse_tags %}
 
 {% for item in menu_items %}
   {% cycle '<ul class="main-menu-items">' '' '' '' %}
   <li>
-    <a href="">{{ item.text }}</a>
+    <a href="{{ item|url_for }}">{{ item.text }}</a>
```
Comment (jacksonj04) [inline cms/templates/partials/megamenu/navigation.html:6]: I wonder if it's worthwhile being more explicit with `{{ url 'browse' item.slug }}`? `url_for` in this context makes me expect the URL for the item itself, rather than browsing it.

File: cms/templates/partials/nhs_header.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{% load frontend_tags menu_tags %}
```
Comment (jacksonj04) [inline cms/templates/partials/nhs_header.html:1]: Is this just a duplicate of the file in `partials/nhs_components/nhs_header.html`?

File: cms/urls.py [hunk 6]
```diff
@@ -1,14 +1,14 @@
 from django.conf import settings
 from django.contrib import admin
 from django.urls import include, path
-
 from wagtail.admin import urls as wagtailadmin_urls
+from wagtail.contrib.sitemaps.views import sitemap
```
Comment (jacksonj04) [inline cms/urls.py:5]: This looks errant

File: docker-compose.yml [hunk 7]
```diff
@@ -15,6 +15,7 @@ services:
       - search
     ports:
       - "8000:8000"
+      - "3000:3000"
```
Comment (jacksonj04) [inline docker-compose.yml:18]: Should this be in this PR? It seems unrelated.

**Comment by jacksonj04** [comment]
A few comments and things to tidy up which seem to have leaked in from other work, but this is looking great! My big worry is that it could really do with some more testing, but @tobybatch I think you're on that?

File: cms/urls.py (hunk 6 above)
Comment (tobybatch) [reply cms/urls.py:5]: That's referenced on line 15

```
path("sitemap.xml", sitemap),
```

It just got moved from line 7. Jet brains has an "optimise imports" macro that re-orders and sorts imports into groups and alpha order.

File: cms/templates/partials/megamenu/navigation.html (hunk 4 above)
Comment (tobybatch) [reply cms/templates/partials/megamenu/navigation.html]: I've renamed it but we can't use the slug lookup because the item is a menu_item not a page. We need to use the link id to look up the item and build the slug.

I've renamed it

**Review by tobybatch — APPROVED** [review]
